site/content/blog/multi-env-deployment-ytt-kapp.md

```diff
@@ -0,0 +1,331 @@
+  labels:
+    app.kubernetes.io/name: sample-app-prod
+  namespace: my-namespace
+spec:
+  selector:
+    app: sample-app-prod
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 8080
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sample-app-prod
+  labels:
+    app.kubernetes.io/name: sample-app-prod
+  namespace: my-namespace
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: sample-app-prod
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: sample-app-prod
+    spec:
+      containers:
+      - name: sample-app
+        image: nginx:1.14.2
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 8080
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: application-settings
+  namespace: my-namespace
+data:
+  app_mode: prod
+  certificates: /etc/ssl/prod
+  db_user: prod-user
+  db_password: prod-password
+```
+
+**Try this** out in our [ytt interactive playground](https://carvel.dev/ytt/#gist:https://gist.github.com/pivotaljohn/495a132fdce5a6889fd5ccf0b48b098f). 
+
+Now let's deploy it using kapp - 
+```bash
+$ kapp deploy -a sample-app -f <(ytt -f app.yaml -f schema.yaml -f values-prod.yaml -f add-namespace.yaml) -y
+Target cluster 'https://127.0.0.1:57418' (nodes: minikube)
+
+Changes
+
+Namespace     Name                  Kind        Conds.  Age  Op      Op st.  Wait to    Rs  Ri  
+my-namespace  application-settings  ConfigMap   -       -    create  -       reconcile  -   -  
+^             sample-app-prod       Deployment  -       -    create  -       reconcile  -   -  
+^             sample-app-prod       Service     -       -    create  -       reconcile  -   -  
+
+Op:      3 create, 0 delete, 0 update, 0 noop, 0 exists
+Wait to: 3 reconcile, 0 delete, 0 noop
+
+6:55:46PM: ---- applying 1 changes [0/3 done] ----
+6:55:46PM: create configmap/application-settings (v1) namespace: my-namespace
+6:55:46PM: ---- waiting on 1 changes [0/3 done] ----
+6:55:46PM: ok: reconcile configmap/application-settings (v1) namespace: my-namespace
+6:55:46PM: ---- applying 2 changes [1/3 done] ----
+6:55:46PM: create deployment/sample-app-prod (apps/v1) namespace: my-namespace
+6:55:46PM: create service/sample-app-prod (v1) namespace: my-namespace
+6:55:46PM: ---- waiting on 2 changes [1/3 done] ----
+...
+6:55:50PM: ---- applying complete [3/3 done] ----
+6:55:50PM: ---- waiting complete [3/3 done] ----
+
+Succeeded
+```
+
+This time I deployed using kapp `-y` flag which will not ask for confirmation before applying the changes. It also shows a progress log while reconciling for the changes to provide details on for which resources it is waiting and what all got applied successfully.
+
+Here, just to limit the scope of this blog I have used some basic but powerful features of ytt for templating and patching but there are many advanced features provided by ytt which will be worth exploring and can match your specific use case.
```

Comment:
```suggestion
Here, just to limit the scope of this article I have used some basic but powerful features of ytt for templating and patching but there are many advanced features provided by ytt which will be worth exploring and can match your specific use case.
```

nit: a blog is a collection of articles.

Reply:
Done. Thanks